src/main/java/gregtech/api/mui/GTGuis.java

```diff
@@ -92,44 +44,48 @@ public static ModularPanel createPanel(ItemStack stack, int width, int height) {
-        return GuiInfo.builder()
-                .clientGui((context, mainPanel) -> {
-                    TileEntity te = context.getTileEntity();
-                    if (te == null) throw new IllegalStateException();
-                    CoverHolder coverHolder = te.getCapability(GregtechTileCapabilities.CAPABILITY_COVER_HOLDER,
-                            facing);
-                    if (coverHolder == null) throw new IllegalStateException();
-                    Cover cover = coverHolder.getCoverAtSide(facing);
-                    if (!(cover instanceof CoverWithUI)) throw new IllegalStateException();
-                    return ((CoverWithUI) cover).createScreen(context, mainPanel);
-                })
-                .commonGui((context, syncHandler) -> {
-                    TileEntity te = context.getTileEntity();
-                    if (te == null) throw new IllegalStateException();
-                    CoverHolder coverHolder = te.getCapability(GregtechTileCapabilities.CAPABILITY_COVER_HOLDER,
-                            facing);
-                    if (coverHolder == null) throw new IllegalStateException();
-                    Cover cover = coverHolder.getCoverAtSide(facing);
-                    if (!(cover instanceof CoverWithUI)) throw new IllegalStateException();
-                    return ((CoverWithUI) cover).buildUI(context, syncHandler, context.getWorld().isRemote);
-                })
-                .build();
+    public static ModularPanel createPopupPanel(String name, int width, int height, boolean disableBelow,
+                                                boolean closeOnOutsideClick) {
+        return new PopupPanel(name, width, height, disableBelow, closeOnOutsideClick);
     }
 
-    private static IGuiHolder getGuiHolder(ItemStack stack) {
-        if (stack.getItem() instanceof MetaItem) {
-            MetaItem<?>.MetaValueItem valueItem = ((MetaItem<?>) stack.getItem()).getItem(stack);
-            if (valueItem != null && valueItem.getUIManager() != null) {
-                return valueItem.getUIManager();
-            }
+    private static class PopupPanel extends ModularPanel {
+
+        private final boolean disableBelow;
+        private final boolean closeOnOutsideClick;
+
+        public PopupPanel(@NotNull String name, int width, int height, boolean disableBelow,
+                          boolean closeOnOutsideClick) {
+            super(name);
+            flex().startDefaultMode();
+            flex().size(width, height).align(Alignment.Center);
+            flex().endDefaultMode();
+            background(GTGuiTextures.BACKGROUND_POPUP);
+            child(new ButtonWidget<>()
+                    .size(10, 10)
+                    .top(5).right(5)
+                    .background(GTGuiTextures.BUTTON) // todo
+                    .overlay(GuiTextures.CROSS_TINY)
+                    .onMousePressed(button -> {
+                        closeIfOpen();
```

Comment:
`closeIfOpen(true)` for animation. Also there is a default panel close button in `ButtonWidget`